### ISSUE TITLE
Idea: package the repo into an container

### ISSUE THREAD

**Issue by koseduhemak** [mag37/dockcheck#77]
Hi there!

Just an idea: why not package the script / repo contents into a container. Then we can use it to run everything inside docker - no need to install dependencies and stuff locally.

* docker socket needs to be mounted in the container
* docker compose needs to be installed (alongside other requirements of this repo)
  * maybe the container can be based on docker/compose image?

What do you think of that?
Best wishes,
koseduhemak

**Comment by mag37** [comment]
It's a great suggestion and I've been considering it myself, just havn't really got the time to spare to lab with building it right now. But I'll try to look into it in the future!

Thank you for the suggestion.

**Comment by mritunjaymusale** [comment]
hi, this seems like a good idea for systems that are immutable for the most part but still let you install apps using docker.

**Comment by mag37** [comment]
> hi, this seems like a good idea for systems that are immutable for the most part but still let you install apps using docker. 

Indeed! I've been meaning to start working on this - just a bunch of other changes first on the list 😅
But I'll try to prioritize this if no issues surface from the latest release.
